storage-api/src/java/org/apache/hadoop/hive/ql/exec/vector/VectorizedRowBatch.java

```diff
@@ -338,6 +339,7 @@ public void write(DataOutput arg0) throws IOException {
    *  - resets each column
    *  - inits each column
    */
+  @Override
   public void reset() {
```

Comment:
Can we also set the selected array to null here?

Reply:
My problem is that VectorizedRowBatch currently has an implicit invariant that selected is never null. Changing that assumption seems potentially problematic.